students/Chris_Hallen/lesson03/mailroom.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+
+
+response = ''
+donor_list = ['Zach Galifianakis', 23757, 3, "Conan O'Brien", 45000, 3, 'Kristen Wiig', 111222, 2, 'Sarah Silverman', 100, 1, 'Bill Burr', 12, 1, 'Will Ferrell', 100000, 2, ]
+
+while response != 'quit':  # This program will constantly run until a user enters 'quit' - regardless of whether 'quit'is upper or lower case or a combination thereof
+    def send_a_thank_you(donor_list):
+        """This function asks whether a user wants to show a list of existing donors, a single existing donor, or
+           to create a new donor.  This function prints a list of existing donors if the user enters 'list' -
+           regardless if 'list' is upper or lower case or a combination thereof.  This function adds a new donor
+           name to the end of the existing donor_list if a donor name entered is not in the donor_list.  If a new donor is entered,
+           this function will then ask how much the new donor wants to contribute.  The new donor's contribution will be added to the
+           end of the existing list along with the amount of times the new donor contributed (i.e. 1).  If
+           an existing donor on the donor_list is selected, the function will ask the user how much the new donor
+           wants to contribute.  The new amount will be added to the grand total the existing donor has already
+           contributed and the number of times the existing donor contributed will be increased by 1."""
+        donor_response = ''
+        donor_response = input("Please enter a full name of a donor.  Enter 'list' to show a full listing of all donors. \n> ")
+        while donor_response == 'list':
+            print(donor_list[::3])
+            donor_response = input("Please enter a full name of a donor. Enter 'list' to show a full listing of all donors.\n> ")
+        member_index = donor_list.index(donor_response) if donor_response in donor_list else -1
+        if member_index >= 0 and member_index < len(donor_list):
+            donation_amount = 0
+            donation_amount = input("Please enter an amount that was donated: ")
+            donation_amount = float(donation_amount)
+            donor_list[member_index + 2] += 1
+            donor_list[member_index + 1] += donation_amount
+            print("\nThank you {} for donation number {} of ${:.2f}!".format(donor_list[member_index], donor_list[member_index + 1], donor_list[member_index + 2]))
+        else:
+            donation_amount = 0
+            donation_amount = input("Please enter an amount that was donated: ")
+            donation_amount = float(donation_amount)
+            donor_list.append(donor_response)
+            donor_list.append(donation_amount)
+            donor_list.append(1)
+            print("\nThank you {} for donation number {} of ${:.2f}!".format(donor_list[-3], donor_list[-1], donor_list[-2]))
+
+    def create_a_report(donor_list):
```

Comment:
I don't see sorting functionality in this report (check assignment instructions please)